### ISSUE TITLE
jsx-no-bind catching all binding instead of just in props

### ISSUE THREAD

**Issue by gerbal** [yannickcr/eslint-plugin-react#640]
`JSX props should not use .bind()  react/jsx-no-bind`

is raised on the following using default settings.

```
render () {
    const boundOnNewRequest = this.onNewRequest.bind(this)
    const boundOnKeyDown = this.onKeyDown.bind(this)

    return (
      <Component
        onNewRequest={boundOnNewRequest}
        onKeyDown={boundOnKeyDown} />
    )
  }
```

**Comment by ljharb** [comment]
It should be raised there - jsx-no-bind is meant to catch all creation of functions in the render path, but *especially* uses of `.bind`, which is particularly slow. You want those in the constructor, not in render.

**Comment by gerbal** [comment]
well, apparently I've been misunderstanding where to bind functions. Thanks :stuck_out_tongue:
